gdbplus/gdb-12.1/gdb/Makefile.in

```diff
@@ -0,0 +1,2615 @@
+    else \
+      echo $${rootme}/../gcc/xg++ -B$${rootme}/../gcc/; \
+    fi; \
+  else \
+    if [ "$(host_canonical)" = "$(target_canonical)" ] ; then \
+      echo $(CXX); \
+    else \
+      t='$(program_transform_name)'; echo g++ | sed -e '' $$t; \
+    fi; \
+  fi`
+
+# The use of $$(x_FOR_TARGET) reduces the command line length by not
+# duplicating the lengthy definition.
+TARGET_FLAGS_TO_PASS = \
+	"prefix=$(prefix)" \
+	"exec_prefix=$(exec_prefix)" \
+	"against=$(against)" \
+	'CC=$$(CC_FOR_TARGET)' \
+	"CC_FOR_TARGET=$(CC_FOR_TARGET)" \
+	"CFLAGS=$(CFLAGS)" \
+	'CXX=$$(CXX_FOR_TARGET)' \
+	"CXX_FOR_TARGET=$(CXX_FOR_TARGET)" \
+	"CXXFLAGS=$(CXXFLAGS)" \
+	"INSTALL=$(INSTALL)" \
+	"INSTALL_PROGRAM=$(INSTALL_PROGRAM)" \
+	"INSTALL_DATA=$(INSTALL_DATA)" \
+	"MAKEINFO=$(MAKEINFO)" \
+	"MAKEHTML=$(MAKEHTML)" \
+	"RUNTEST=$(RUNTEST)" \
+	"RUNTESTFLAGS=$(RUNTESTFLAGS)" \
+	"FORCE_PARALLEL=$(FORCE_PARALLEL)" \
+	"TESTS=$(TESTS)"
+
+# All source files that go into linking GDB.
+
+# Files that should wind up in SFILES and whose corresponding .o
+# should be in COMMON_OBS.
+COMMON_SFILES = \
+	ada-lang.c \
+	ada-tasks.c \
+	ada-typeprint.c \
+	ada-valprint.c \
+	ada-varobj.c \
+	addrmap.c \
+	agent.c \
+	alloc.c \
+	annotate.c \
+	arch-utils.c \
+	async-event.c \
+	auto-load.c \
+	auxv.c \
+	ax-gdb.c \
+	ax-general.c \
+	bcache.c \
+	bfd-target.c \
+	block.c \
+	blockframe.c \
+	break-catch-exec.c \
+	break-catch-fork.c \
+	break-catch-sig.c \
+	break-catch-syscall.c \
+	break-catch-throw.c \
+	breakpoint.c \
+	bt-utils.c \
+	btrace.c \
+	build-id.c \
+	buildsym-legacy.c \
+	buildsym.c \
+	c-lang.c \
+	c-typeprint.c \
+	c-valprint.c \
+	c-varobj.c \
+	charset.c \
+	cli-out.c \
+	coff-pe-read.c \
+	coffread.c \
+	complaints.c \
+	completer.c \
+	copying.c \
+	corefile.c \
+	corelow.c \
+	cp-abi.c \
+	cp-namespace.c \
+	cp-support.c \
+	cp-valprint.c \
+	ctfread.c \
+	d-lang.c \
+	d-namespace.c \
+	d-valprint.c \
+	dbxread.c \
+	dcache.c \
+	debug.c \
+	decode.c \
+	debuginfod-support.c \
+	dictionary.c \
+	disasm.c \
+	displaced-stepping.c \
+	dummy-frame.c \
+	dwarf2/abbrev.c \
+	dwarf2/attribute.c \
+	dwarf2/comp-unit-head.c \
+	dwarf2/cu.c \
+	dwarf2/dwz.c \
+	dwarf2/expr.c \
+	dwarf2/frame-tailcall.c \
+	dwarf2/frame.c \
+	dwarf2/index-cache.c \
+	dwarf2/index-common.c \
+	dwarf2/index-write.c \
+	dwarf2/leb.c \
+	dwarf2/line-header.c \
+	dwarf2/loc.c \
+	dwarf2/macro.c \
+	dwarf2/read.c \
+	dwarf2/section.c \
+	dwarf2/stringify.c \
+	eval.c \
+	event-top.c \
+	exceptions.c \
+	exec.c \
+	expprint.c \
+	extension.c \
+	f-lang.c \
+	f-typeprint.c \
+	f-valprint.c \
+	filename-seen-cache.c \
+	filesystem.c \
+	findcmd.c \
+	findvar.c \
+	frame.c \
+	frame-base.c \
+	frame-unwind.c \
+	gcore.c \
+	gdb-demangle.c \
+	gdb_bfd.c \
+	gdb_dep.c \
+	gdbtypes.c \
+	gmp-utils.c \
+	gnu-v2-abi.c \
+	gnu-v3-abi.c \
+	go-lang.c \
+	go-typeprint.c \
+	go-valprint.c \
+	heap.c \
+	heap_ptmalloc_2_35.c \
```

Comment:
Should we add the heap_ptmalloc.c and heap_tcmalloc.c here? Otherwise, the code might not compile.

Reply:
It seems the pt 2.27 or before is not supported anymore as in this issue https://github.com/yanqi27/core_analyzer/issues/37, I'm still trying to think how to keep all the supported heap can live tegother.